src/ridepy/util/dispatchers/ridepooling.py

```diff
@@ -386,3 +407,202 @@ def BruteForceTotalTravelTimeMinimizingDispatcher(
+                )
+                # Check time violations?
+                # Hier muss ueberprueft werden, ob druch das einfügen irgendwo die seat capacity verletzt wird
+                stoplist_request_in_vehicle = stoplist[
+                    best_pickup_idx : best_dropoff_idx + 1
+                ]
+                # occupancies_ausschnitt = list(map(lambda x: x.occupancy_after_servicing, stoplist_request_in_vehicle))
+
+                occupancies_ausschnitt = []
+                for x in stoplist_request_in_vehicle:
+                    occupancies_ausschnitt.append(x.occupancy_after_servicing)
+
+                if seat_capacity in occupancies_ausschnitt:
+                    best_pickup_idx = len(stoplist) - 1
+                    best_dropoff_idx = len(stoplist) - 1
+                    continue
+                if CPAT_do > request.delivery_timewindow_max:
+                    best_pickup_idx = len(stoplist) - 1
+                    best_dropoff_idx = len(stoplist) - 1
+                    continue
+                boolInsertEnd = False
+                min_cost = CPAT_do
+                break
+        else:
+            continue
+
+    if boolInsertEnd:
+        best_pickup_idx = len(stoplist) - 1
+        best_dropoff_idx = len(stoplist) - 1
+        time_to_pickup = space.t(stoplist[best_pickup_idx].location, request.origin)
+        CPAT_pu = cpat_of_inserted_stop(stoplist[best_pickup_idx], time_to_pickup)
+        EAST_pu = request.pickup_timewindow_min
+        CPAT_do = max(EAST_pu, CPAT_pu) + space.t(request.origin, request.destination)
+        if CPAT_pu > request.pickup_timewindow_max:
+            min_cost = np.inf
+        elif CPAT_do > request.delivery_timewindow_max:
+            min_cost = np.inf
+        else:
+            min_cost = CPAT_do
+
+    if min_cost < np.inf:
+        new_stoplist = insert_request_to_stoplist_drive_first(
+            stoplist=stoplist,
+            request=request,
+            pickup_idx=best_pickup_idx,
+            dropoff_idx=best_dropoff_idx,
+            space=space,
+        )
+        EAST_pu, LAST_pu = (
+            new_stoplist[best_pickup_idx + 1].time_window_min,
+            new_stoplist[best_pickup_idx + 1].time_window_max,
+        )
+        EAST_do, LAST_do = (
+            new_stoplist[best_dropoff_idx + 2].time_window_min,
+            new_stoplist[best_dropoff_idx + 2].time_window_max,
+        )
+
+        listOccupanciesNewStopList = list(
+            map(lambda x: x.occupancy_after_servicing, new_stoplist)
+        )
+        for item in listOccupanciesNewStopList:
+            if item > seat_capacity:
+                print("Seat capacity error!!")
+
+        for item in new_stoplist[1:]:
+            request_item = item.request
+            if item.action.name == "pickup":
+                if item.estimated_arrival_time > request_item.pickup_timewindow_max:
+                    print("Time violation!!")
```

Comment:
As above: I don't understand this print statement. Normally, the dispatcher should not print anything. If indeed an unexpected error happens, this should be raised.